Apply consistent font sizes and spacing in checkout (miscellaneous small blocks)

straku opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Closes [WOOPLUG-5061](https://linear.app/a8c/issue/WOOPLUG-5061/apply-consistent-font-sizes-and-spacing-in-checkout-miscellaneous)
Closes [WOOPLUG-5151](https://linear.app/a8c/issue/WOOPLUG-5151/some-elements-from-the-contact-information-block-are-still-using)

This PR covers changes for the following blocks:

<img width="229" height="134" alt="Screenshot 2025-08-18 at 16 10 06" src="https://github.com/user-attachments/assets/aee597e0-b305-4e0c-b873-9ecbc6a3db3c" />

Additionally:
- the issue with relative font-size in the Customer Information block is fixed
- further improvements for mobile are applied (reduced spacing between checkout steps and no border)

### Screenshots or screen recordings:

#### Order note and terms on mobile (Pixel 7)

| Before | After |
|--------|-------|
| <img width="412" height="354" alt="CleanShot 2025-08-18 at 15 43 31@2x" src="https://github.com/user-attachments/assets/dfefdb7b-cb0f-4cb6-8324-59fdbd03d167" /> | <img width="412" height="169" alt="CleanShot 2025-08-18 at 15 40 00@2x" src="https://github.com/user-attachments/assets/eeb01112-4cd0-4ed8-98d4-65d33dbf52f0" /> |

#### Actions on mobile (Pixel 7)

| Before | After |
|--------|-------|
| <img width="824" height="322" alt="CleanShot 2025-08-18 at 15 43 50@2x" src="https://github.com/user-attachments/assets/c7b59463-6183-4ede-abd9-14e10e276c44" /> | <img width="824" height="366" alt="CleanShot 2025-08-18 at 15 40 19@2x" src="https://github.com/user-attachments/assets/b81dbc0f-5570-4405-9475-71550d2a5886" /> |

#### Order note, terms, and actions on desktop

| Before | After |
|--------|-------|
| <img width="1620" height="520" alt="CleanShot 2025-08-18 at 15 42 53@2x" src="https://github.com/user-attachments/assets/1b9cc71d-21be-49ae-9d48-291f936f4c22" /> | <img width="1620" height="452" alt="CleanShot 2025-08-18 at 15 41 00@2x" src="https://github.com/user-attachments/assets/449d3948-001c-4075-8318-e0871eea62b3" /> |

#### And a small fix to the Contact Information block

Not using relative font size anymore.

| Before | After |
|--------|-------|
| <img width="412" height="249" alt="Screenshot 2025-08-18 at 15 38 44" src="https://github.com/user-attachments/assets/029d10a3-2b11-45fa-bec7-ff49db8664db" /> | <img width="412" height="240" alt="Screenshot 2025-08-18 at 15 39 04" src="https://github.com/user-attachments/assets/ce745011-16e4-461f-8c11-3b08587747a3" /> |

### How to test the changes in this Pull Request:

Since we're covering the "Additional Information" block, we need some custom fields there to see it. Use this snippet to set it up.

```php
add_action(
	'init',
	function () {
		woocommerce_register_additional_checkout_field(
			array(
				'id'          => 'namespace/how-did-you-hear-about-us',
				'label'       => 'How did you hear about us?',
				'placeholder' => 'Select a source',
				'location'    => 'order',
				'type'        => 'select',
				'options'     => [
					[
						'value' => 'google',
						'label' => 'Google'
					],
					[
						'value' => 'facebook',
						'label' => 'Facebook'
					],
					[
						'value' => 'friend',
						'label' => 'From a friend'
					],
					[
						'value' => 'other',
						'label' => 'Other'
					],
				]
			)
		);
	}
);
```

Now, for the steps:
1. Set up the snippet presented above (or any other custom field that will be located within the "Additional Information" block)
2. Add some items to the cart and proceed to checkout
3. Verify that the modified blocks are aligned with the designs and that you don't see anything out of place
4. Test these "non-standard" flows:
  - Click "add a note" and make sure the textarea spacing and typography is aligned with other text inputs
  - Open editor, focus on "Terms and Conditions" block and enable "Require checkbox" and "Show separator" attributes. Verify that they are applied properly and look aligned with the rest of the page (designs are not covering those settings)
  - As above, but go to "Actions" and change "Hide price" to "Show price"; also play with disabling the display of return to cart link.
  - As above, but go to "Checkout" top-level block and enable "Show form step numbers".

Test for the Contact Information fix:
1. Make sure that your store has guest checkout enabled
2. Edit global styles of your page in the editor; go to text and increase the default size of the text to "L" (or anything larger than the default)
3. Open your store in an incognito window
4. Add some items to the cart and proceed to checkout
5. Make sure that the text "You are currently checking out as a guest." below the email address input is not bigger than the surrounding texts.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.
-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment

This PR targets the feature branch. A changelog will be created when this feature branch gets merged to trunk.

</details>

straku committed:
Apply consistent font sizes and spacing in checkout (payment methods)

straku committed:
FormStep: Don't include the heading container element it it's empty

straku committed:
Remove redundant type

straku committed:
Fix usage of `useSelect`

straku committed:
Order notes change

straku committed:
Add mixin for button font size

straku committed:
Remove redundant fragment

straku committed:
Remove bottom borders between steps on mobile

straku committed:
Remove built in margin in checkbox

straku committed:
Cover rest of the checkout blocks

straku committed:
Fix WOOPLUG-5151

straku committed:
Fix terms when separator is enabled

ralucaStan reviewed (changes requested):
I added some comments from my smoke testing; I didn't check the code, I'll do that after the comments below are addressed.

straku committed:
Fix TypeScript error

straku committed:
Add space around terms checkbox in the editor

straku committed:
Fix unnecessary full width on place order button + refactor actions block in editor

straku committed:
Remove layout shift for note textarea

straku commented:
@ralucaStan Reporting back 🫡 

> there are some spaces missing for the Billing and Payments areas

I can't reproduce this. I hope it fixed itself upon rebasing with trunk. If not, let me know, and I will take a deeper look.

> on desktop I see a border between the Add order section and the content below; I think it's coming from the terms block, see 2nd image

This is expected if we want to retain this option in the terms block:
<img width="552" height="252" alt="CleanShot 2025-08-25 at 18 36 19@2x" src="https://github.com/user-attachments/assets/a6c0cef0-54d3-4831-b520-6b3d4619bc2a" />

> writing on the order note and bluring the textarea creates some vertical content shift

Fixed in https://github.com/woocommerce/woocommerce/pull/60402/commits/2d2d62daeb049eadb1fe060a2067d42e55772447

> Terms checkbox in the editor needs spacing before the text (only in the editor)

Fixed in https://github.com/woocommerce/woocommerce/pull/60402/commits/39af2f4a19e25148f038dcfc7d814429cbbfe0c3

> Hiding the Return to cart block from the Actions inner block creates a full width Place order button. Is this expected?

It was expected, but new designs don't have this behavior. I removed it in https://github.com/woocommerce/woocommerce/pull/60402/commits/95126ab8e89d4d97bb578a0ac63ddc94f4593ded. I noticed extensive duplication around this logic refactored it a bit to remove it. I would appreciate a deeper testing around this if you can 🙏

ralucaStan commented on plugins/woocommerce/client/blocks/packages/components/textarea/style.scss:
```diff
@@ -1,30 +1,28 @@
-		border-color: $input-border-dark;
-		color: $input-text-dark;
-
-		&::placeholder {
```
On the Checkout block you can select Dark mode inputs, used for themes with a dark background color. You'll see that the textarea now has a different placeholder color color then the rest of the form elements

straku replied:
Thanks for catching, fixed.

ralucaStan commented on plugins/woocommerce/client/blocks/packages/components/form-step/style.scss:
```diff
@@ -13,47 +13,6 @@
 	.wc-block-components-country-input {
 		margin-top: 0;
 	}
-
```
The layout of the steps has changed since I last reviewed, notice the line is on the right. It should be under the number. 

<img width="791" height="363" alt="Screenshot 2025-08-26 at 17 38 30" src="https://github.com/user-attachments/assets/2242cef0-8ab6-43cd-9997-93a0ca899c0e" />

Before: 
<img width="781" height="277" alt="Screenshot 2025-08-26 at 17 43 23" src="https://github.com/user-attachments/assets/491e0bbf-8e28-4ee7-9c87-3588c81d54ad" />

ralucaStan commented on plugins/woocommerce/client/blocks/assets/js/blocks/checkout/inner-blocks/checkout-actions-block/block.tsx:
```diff
@@ -59,28 +54,21 @@ const Block = ( {
 					</ReturnToCartButton>
 				) }
-				{ showPrice && (
-					<style>
```
👍

ralucaStan commented on plugins/woocommerce/client/blocks/assets/js/blocks/checkout/inner-blocks/checkout-order-note-block/block.tsx:
```diff
@@ -22,7 +22,8 @@ const Block = ( { className }: { className?: string } ): JSX.Element => {
 			};
-		}
+		},
+		[]
```
👍

ralucaStan commented on plugins/woocommerce/client/blocks/assets/js/blocks/checkout/inner-blocks/checkout-terms-block/edit.tsx:
```diff
@@ -145,10 +145,9 @@ export const Edit = ( {
 				{ checkbox ? (
 					<>
-						<CheckboxControl
-							id="terms-condition"
```
I think we might need to keep this id for people using it in their customizations.

ralucaStan reviewed (changes requested):
Hey @straku, the initial bugs I've mentioned have been fixed, thank you for looking into them. I added another review.


 I also noticed that this Order summary title is missing some spacing on mobile. We render the order summary differently on mobile, so I guess you need to handle it separately. 
 
<img width="188" height="333" alt="Screenshot 2025-08-26 at 17 21 45" src="https://github.com/user-attachments/assets/fb56f5ec-b748-459f-99b8-8a2dcfc97381" />

I might be AFK when the changes are addressed, so feel free to assign to somebody else.

Nice work!

straku committed:
Fix border position when steps are enabled

straku committed:
Align textarea placeholder colors with text-input label

straku committed:
Bring back terms checkbox id

straku commented:
@opr Since Raluca is AFK, can you review this one? 🙏

opr commented on plugins/woocommerce/client/blocks/packages/components/form-step/style.scss:
It seems the changes here have affected the second order summary (at the bottom of the page) on mobile, the heading is now too close to the summary

<img width="412" height="108" alt="Image" src="https://github.com/user-attachments/assets/f19e2a60-bc25-48c9-b30c-048a7f762793" />

straku replied:
Already noticed by @ralucaStan, since I intend to cover this section fully next, I decided to handle it in the follow-up PRs.

opr commented on plugins/woocommerce/client/blocks/packages/components/textarea/style.scss:
```diff
@@ -21,10 +34,3 @@
 	}
 }
-
-.theme-twentytwentyone {
-	.has-dark-controls .wc-block-components-textarea {
-		background-color: $input-background-dark;
-		color: $input-text-dark;
-	}
-}
```
Why was this removed? When editing the textarea in TT1 the font appears in a different color now, was this intentional?

straku replied:
It was removed to align the textarea with the text inputs that we have in the same form. It's weird to have conceptually the same component styled differently depending on the placement in the form.

opr replied:
But when this code is removed, the text becomes black in TT1 while editing it

<img width="418" height="383" alt="image" src="https://github.com/user-attachments/assets/de1b60e2-6f4d-4caa-917c-cd0658d95259" />

When blurred, it looks the same, but I think this should be re-added.

straku replied:
It's a miss on my side then. I am going to fix it, but not by reverting the code to the previous one, but by making sure ad hoc fixes for themes are aligned between text input and textarea. Does it sound good?

opr replied:
Sounds good! 🫡

straku replied:
Ended up just bringing them back; it was a specificity issue (theme styles had higher specificity). Change: https://github.com/woocommerce/woocommerce/pull/60402/commits/31a5da62dff48c5001245a0e511780476ffbc032

opr commented on plugins/woocommerce/client/blocks/packages/components/checkbox-control/index.tsx:
```diff
@@ -13,7 +13,7 @@ import './style.scss';
 export type CheckboxControlProps = {
 	label?: string | React.ReactNode;
-	onChange: ( value: boolean ) => void;
+	onChange?: ( value: boolean ) => void;
```
Where was the TS error? Should these inputs not always have an `onChange`?

straku replied:
We didn't use `onChange` when `CheckboxControl` is used on the editor. I can make it required and provide a `noop` function there if you prefer.

opr replied:
I think making it required and passing a noop makes more sense, thanks.

straku replied:
Changed in https://github.com/woocommerce/woocommerce/pull/60402/commits/9d6a0982178e14e6d4db9fcd6ae99b8ea48af94e

opr commented on plugins/woocommerce/client/blocks/assets/js/blocks/checkout/inner-blocks/checkout-terms-block/frontend.tsx:
```diff
@@ -90,24 +90,20 @@ const FrontendBlock = ( {
-							/>
-						</CheckboxControl>
-					</>
+					<CheckboxControl
```
Minor nitpick, is it possible that this could regress third party CSS, I know we don't guarantee DOM structure, but I feel this could potentially introduce issues.

straku replied:
Yes, true. But I already did that a few times while working on this project. If you feel strongly about making sure it's the same, I need to do a pass over the full branch. I am worried it won't be possible everywhere, though.

tjcafferkey replied:
I haven't verified in this case but fragments `<></>` don't return `<div>` elements anyway. They were added to React to prevent people unnecessarily introducing `<div>` elements when JSX required a parent node.

opr replied:
True, I think this is fine then @straku 👍🏼

opr commented on plugins/woocommerce/client/blocks/assets/js/blocks/checkout/inner-blocks/checkout-actions-block/edit.tsx:
```diff
@@ -134,50 +133,28 @@ export const Edit = ( {
-							</>
-						) }
-					</EditableButton>
+					<PlaceOrderButton
```
Nice, we should apply this to the other places `EditableButton` is used in a followup and remove it

straku replied:
Noted 👍

opr reviewed (changes requested):
Hi @straku thanks for this work - I was testing with various themes and browsers, and Firefox, Edge, Safari all seem to be working fine with a couple of minor nitpicks I mentioned in the review.

One thing I noticed, that I don't think is caused by this PR, but highlighting it just to make sure: on a dark theme, the saved address card is all but unreadable, and the lines between the numbers when step numbers are enabled are also too dark. 

<img width="496" height="406" alt="Image" src="https://github.com/user-attachments/assets/622f502e-d87d-402a-ac40-965176cc7532" />

The separators between line items and the T&C block also suffer from this.

Also, in your testing instructions you note:

> Make sure that the text "You are currently checking out as a guest." below the email address input is bigger than the surrounding texts.

Do you mean is _not_ bigger? Just checking.

straku committed:
CheckboxControl: Make `onChange` required and cleanup types

straku committed:
Bring back specific styles for tt1

straku committed:
Fix address card colors on dark backgrounds

straku commented:
@opr

> One thing I noticed, that I don't think is caused by this PR, but highlighting it just to make sure: on a dark theme, the saved address card is all but unreadable, and the lines between the numbers when step numbers are enabled are also too dark.

I addressed this in https://github.com/woocommerce/woocommerce/pull/60402/commits/fc731f2d302086384a70e9afb33e9d7e153a30cd. There was already a precedent of using this color for a similar purpose in checkout. When it comes to borders for steps and terms, I don't see a great match. Will ask Yoann.

straku committed:
Remove unnecessary import

straku committed:
Fix tests

opr reviewed (approved):
Thanks for the changes @straku - I see you plan to pick up the changes to the order summary in a follow up, so I won't let this block the PR.

Happy for this to be merged now I've had a look at everything in Edge, Firefox, and Safari and desktop/mobile views.
